src/partitioner/local/RDFParser.cpp

```diff
@@ -0,0 +1,422 @@
+                                attributes[2] = journal_name;
+
+
+                            } else if (XMLString::equals(currentElement2->getTagName(), TAG_has_volume)) {
+
+                                const XMLCh *vol = currentElement2->getTextContent();
+
+                                volume = XMLString::transcode(vol);
+                                attributes[6] = volume;
+
+
+                            } else if (XMLString::equals(currentElement2->getTagName(), TAG_has_date)) {
+
+
+                                DOMNode *child = currentElement2->getFirstElementChild();
+                                DOMElement *childElement
+                                        = dynamic_cast< xercesc::DOMElement * >( child );
+
+
+                                DOMNodeList *dateElements = childElement->getChildNodes();
+                                const XMLSize_t elementCount = dateElements->getLength();
+
+
+                                for (XMLSize_t e = 0; e < elementCount; ++e) {
+
+                                    DOMNode *dateEle = dateElements->item(e);
+                                    if (dateEle->getNodeType() &&
+                                        dateEle->getNodeType() == DOMNode::ELEMENT_NODE) // is element
+                                    {
+
+
+                                        DOMElement *currentDateElement = dynamic_cast< xercesc::DOMElement * >( dateEle );
+                                        if (XMLString::equals(currentDateElement->getTagName(), TAG_year_of)) {
+
+                                            const XMLCh *yearOf = currentDateElement->getTextContent();
+                                            year = XMLString::transcode(yearOf);
+                                            attributes[5] = year;
+
+                                        } else if (XMLString::equals(currentDateElement->getTagName(), TAG_month_of)) {
+
+                                            const XMLCh *monthOf = currentDateElement->getTextContent();
+                                            month = XMLString::transcode(monthOf);
+                                            attributes[4] = month;
+
+                                        }
+                                    }
+                                }
+
+
+                            } else if (XMLString::equals(currentElement2->getTagName(), TAG_has_web_address)) {
+
+                                const XMLCh *web_add = currentElement2->getTextContent();
+                                web_address = XMLString::transcode(web_add);
+                                attributes[3] = web_address;
+
+
+                            }
+
+                        }
+                    }
+
+
+                }
+
+                if (attributes->size() > 0) {
+                    articlesMap.insert({articleID, attributes});
+
+                }
+                for (int i = 0; i < authorsInArticle.size(); i++) {
```

Comment:
Isn't there a way to remove these two for loops?

Reply:
This map contains authors for one article. I need make all possible combinations to make edges among them. To do that I needed two loops.

Reply:
Ok